### ISSUE TITLE
So given a SHA I can't find out what branch it's from without parsing back to it from a known branch

### ISSUE THREAD

**Issue by rogernolan** [githubteacher/advanced-apr-2015#11]

**Comment by pcharlan** [comment]
That's correct.  Commits aren't 'on' branches in the same sense they are in some (most?) other SCMs, they're just reachable from branch references, which point to the tips of history.

**Comment by pcharlan** [comment]
You can use Git 'notes' to record which branch a commit was pushed to if you want that data preserved, but that information isn't part of Git's history model.

**Comment by githubteacher** [comment]
@rogernolan Did @pcharlan's answers and Joshua's explanation of this vis-à-vis the GitHub UI answer your question?

:tophat: tip to @pcharlan for contributing so many answers!
